### PR TITLE
Makefile improvements

### PR DESCRIPTION
**Pull request by couchand**
Adds a check target to run `cargo check --all-features`, which I've found useful to verify I'm not totally breaking things.

Also adds `all` and `clean` to the `.PHONY` list.

### CODE REVIEW

**Comment by couchand** [comment]
Added `rm src/generic.rs` to the clean target.  Probably not necessary generally since it will get clobbered, but as I'm hacking around on `svd2rust` I want to be absolutely sure it's getting the new version, and it seems like the right thing to do anyway.

File: Makefile [hunk 1]
```diff
@@ -56,11 +56,17 @@ macros/src/vector.rs: src/devices/*/interrupt.rs
 	@echo -e "\tGEN-VECTOR\t>macros/src/vector.rs"
 	@./gen-intr-lut.sh $^ >$@
 
+check: all
+	@echo -e "\tCHECK"
+	@cargo check --all-features
+
```
Comment (Rahix) [inline Makefile:62]: I'd rather keep the makefile just for generating the files, not for actually invoking any kind of compilation, linting, etc.

Comment (couchand) [reply Makefile:62]: That's a fair point.

I only added it after forgetting to pass `--all-features` and hitting the compile warning one too many times.

Comment (couchand) [reply Makefile]: It's gone.

**Comment by Rahix** [comment]
Thank you very much!

**Pull request merged**